crates/loader/src/local.rs

```diff
@@ -135,6 +136,21 @@ impl LocalLoader {
         })
     }
 
+    fn env_checker((key, val): (String, Variable)) -> anyhow::Result<(String, Variable)> {
+        if val.default.is_none() {
+            if std::env::var(env_key(None, key.as_ref())).is_err() {
```

Comment:
The environment provider appears to support a `dotenv_path` setting to merge values from a file - not clear that this is respected here?